### ISSUE TITLE
Create Packer and/or Vagrant scripts for install

### ISSUE THREAD

**Issue by day1player** [mandiant/commando-vm#136]

**Comment by day1player** [comment]
As discussed in #59

**Comment by joeminicucci** [comment]
This would be really nice to have!

**Comment by day1player** [comment]
Due to Windows Tamper Protection I am not sure this will be a viable option for install.

**Comment by elreydetoda** [comment]
I haven't looked at the code for this yet, but it looks promising: https://github.com/clong/DetectionLab/pull/307


from, this issues comments: https://github.com/clong/DetectionLab/issues/306

**Comment by day1player** [comment]
Awesome, thanks again for the help ;-)

**Comment by elreydetoda** [comment]
no problem :grin: glad to help where I can. thanks @clong for amazing dev work in DetectionLab.

**Comment by DrJZoidberg** [comment]
@day1player there are a few different ways we can take the packer scripts. 

1. take the template from iso to full machine w/o waiting for snapshots in between. i.e. A-> Z

This would result in 1 VM with 1 final state.

2. have incremental templates which would do something like so:
  a. iso -> "clean-ish" machine, which has some required changes to run the commando install script but not much more (disabling tamper protection for instance) A->M
  b. "clean-ish" -> final produced build. M->Z

This would result in 1 VM with 2 final states (snapshots), **or**
2 VMs with 1 state each (the b. step above would start by making a clone of the existing "clean-ish" VM to use). 

Which do you think you would prefer?

Additionally, which builder targets do you need? right now I'm planning to use virtualbox, vmware-workstation, and hyper-v.

**Comment by NebulousAnchor** [comment]
@DrJZoidberg I was about to start building packer/vagrant versions tomorrow for a project I am working on to make a quick deployment kit using RHEL Satellite server and Ansible/Vagrant.  I anticipate the tamper protection being an issue for a single command packer build.  However, I will pass along any lessons learned and the files here if I manage to find a way to do it. 

As far as builder targets, I personally will be aiming for libvirt since my host OS for the deploy-able devices will be RHEL.

**Comment by NebulousAnchor** [comment]
So I found a way to help automate all of the install BUT the final step where I have to type 'ENTER'  

In the packer .json file I have a setup.ps1 file, the commando install.ps1 file, the AutoHotKey.exe portable exe, and the attached file (rename from .txt to .ahk) forked onto the box using the file provisioner.

The setup.ps1:

AutoHotKey.exe disable_defender.ahk
Set-ExecutionPolicy Unrestricted
Unblock-File .\install.ps1
.\install.ps1

[disable_defender.txt](https://github.com/fireeye/commando-vm/files/5077191/disable_defender.txt)

**Comment by NebulousAnchor** [comment]
@DrJZoidberg  I need to redo my packer tomorrow.  I'll try your files.

**Comment by day1player** [comment]
https://twitter.com/jonasLyk/status/1293815234805760000

**Comment by skollr34p3r** [comment]
Create a new issue here if you are still looking to see this happen in CommandoVM: https://github.com/mandiant/VM-Packages/issues/new/choose
